services/storage/virtualharddisk/wssd.go

```diff
@@ -79,6 +80,38 @@ func (c *client) Delete(ctx context.Context, group, container, name string) erro
 
 }
 
+func (c *client) Precheck(ctx context.Context, group, container string, vhds []*storage.VirtualHardDisk) (bool, error) {
+	request, err := getVirtualHardDiskPrecheckRequest(group, container, vhds)
+	if err != nil {
+		return false, err
+	}
+
+	response, err := c.VirtualHardDiskAgentClient.Precheck(ctx, request)
+	if err != nil {
+		return false, err
+	} else {
+		result := response.Result.GetValue()
```

Comment:
Can we refactor this result code to a seperate functtion getVirtualHardDiskPrecheckRespose

Reply:
OK. Also refactor the one for VM.